### PR TITLE
Add retry method to share link dialog in case of error

### PR DESCRIPTION
**Pull request by cmdcolin**
example screenshot of new error box with the refresh icon

image

changes the text "Show stack trace" to the "Report" icon, and optionally displays a refresh icon if a "onReset" callback is passed to ErrorMessage

fixes https://github.com/GMOD/jbrowse-components/issues/4274

### CODE REVIEW

File: packages/core/ui/ErrorMessageStackTraceDialog.tsx [hunk 1]
```diff
@@ -7,32 +7,62 @@ import {
...
+  const currentScriptContent = await myfetchtext(uri)
 
   let mapUri =
     new RegExp(/\/\/# sourceMappingURL=(.*)/).exec(currentScriptContent)?.[1] ||
     ''
   mapUri = new URL(mapUri, uri).href + uriQuery
 
-  const map = await (await fetch(mapUri)).json()
-
+  const map = new SourceMapConsumer(await myfetchjson(mapUri))
```
Comment (cmdcolin) [inline packages/core/ui/ErrorMessageStackTraceDialog.tsx:64]: this is a bonus optimizatiom. if the stack trace goes through 20 stack frames of mobx-state-tree function calls, then it was reparsing the stack trace 20 times which added up to 5 seconds of waiting for the stack trace. this parses it just once

**Comment by carolinebridge-oicr** [comment]
Bit jarring how the ui "disappears" when you view the stack trace, I think I'd like to see the stack trace inline with the rest of the modal. At a minimum, provide the user the retry button so they don't have to close + reopen.

image

-- also, when there's a failure to fetch copy to clipboard copies nothing, but provides a success message. Maybe disable it until there's something ready to copy?

image

**Comment by cmdcolin** [comment]
>Bit jarring how the ui "disappears" when you view the stack trace, I think I'd like to see the stack trace inline with the rest of the modal. At a minimum, provide the user the retry button so they don't have to close + reopen.

not sure i understand, do you have a screen recording? note that chunk loading errors are a particular case of this that may be tricky to add retry functionality for https://github.com/GMOD/jbrowse-components/issues/3523

**Comment by carolinebridge-oicr** [comment]
> not sure i understand, do you have a screen recording? note that chunk loading errors are a particular case of this that may be tricky to add retry functionality for https://github.com/GMOD/jbrowse-components/issues/3523

You can see it in the screenshot -- the button options to bookmark, copy, close disappear

**Comment by cmdcolin** [comment]
gotcha. possibly adding another "ErrorBoundary" could catch it but i'm not sure where. the error boundary you are seeing in that screenshot is likely this one https://github.com/GMOD/jbrowse-components/blob/3e6519a6c9d444dc4a18ec5b97e4c4145513715e/packages/core/ui/Dialog.tsx#L71-L88

**Pull request merged**